src/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -49,7 +49,7 @@ void smt2_incremental_decision_proceduret::define_dependent_functions(
         return expr_identifier.first;
       });
   std::stack<exprt> to_be_defined;
-  const auto dependencies_needed = [&](const exprt &expr) {
+  const auto push_dependencies_needed = [&](const exprt &expr) {
```

Comment:
Can the type of the return value of the lambda also be defined there? `-> bool` - because it made it surprising to see a lambda named `push-*` in a context that demands a predicate without it having an explicit book return value.